### PR TITLE
Schema to Contract methods

### PR DESCRIPTION
**Pull request by udit-gulati**

### CODE REVIEW

File: packages/polar/src/lib/deploy/contract.ts [hunk 1]
```diff
@@ -9,60 +9,173 @@ import {
...
+      console.error(`Invalid ${msgName} call. Argument count ${args.length}, expected ${argNames.length}`);
+      return;
+    }
+
+    const msgArgs: any = {};
+    argNames.forEach((abiParam, i) => {
+      msgArgs[abiParam.name] = args[i];
+    });
+
+    // Query the current count
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:41]:
```suggestion
    // Query function
```

File: packages/polar/src/lib/deploy/contract.ts [hunk 2]
```diff
@@ -9,60 +9,173 @@ import {
...
+    }
+
+    const account: Account = (args[args.length - 1] as Account);
+
+    const msgArgs: any = {};
+    argNames.forEach((abiParam, i) => {
+      msgArgs[abiParam.name] = args[i];
+    });
+
+    // Query the current count
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:75]:
```suggestion
    // Execute function (write)
```

File: packages/polar/src/lib/deploy/contract.ts [hunk 3]
```diff
@@ -9,60 +9,173 @@ import {
...
+
   constructor (contractName: string, env: PolarRuntimeEnvironment) {
-    this.contractName = contractName;
+    this.contractName = contractName.replace('-', '_');
     this.codeId = 0;
     this.contractCodeHash = "mock_hash";
-    this.contractAddress = "mock_address";
-    this.contractPath = path.join(ARTIFACTS_DIR, "contracts", `${contractName}.wasm`);
-    this.schemaPath = path.join(SCHEMA_DIR, `${contractName}.json`);
+    this.contractAddress = "secret15l3z73lvy0357qrnyhcfpcxhdrxc209wpvkayl";
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:108]: could you please update this to mock_address ?

File: packages/polar/src/lib/deploy/contract.ts [hunk 4]
```diff
@@ -9,60 +9,173 @@ import {
...
+    this.executeSchemaPath = path.join(SCHEMA_DIR, this.contractName, "handle_msg.json");
+
+    if (!fs.existsSync(this.initSchemaPath)) {
+      console.log("Warning: Init schema not found for contract ", chalk.cyan(contractName));
+    }
+    if (!fs.existsSync(this.querySchemaPath)) {
+      console.log("Warning: Query schema not found for contract ", chalk.cyan(contractName));
+    }
+    if (!fs.existsSync(this.executeSchemaPath)) {
+      console.log("Warning: Execute schema not found for contract ", chalk.cyan(contractName));
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:122]: should we throw error here ?

File: packages/polar/src/types.ts [hunk 5]
```diff
@@ -337,3 +337,41 @@ export type PromiseAny = Promise<any>;
...
+//   from: string;
+//   txHash?: string;
+//   blockHash?: string;
+//   // error?: {
+//   //   message?: any;
+//   //   data?: any;
+//   // };
+//   // result: SubmittableResult;
+//   // events?: DecodedEvent[];
+// }
```
Comment (coco-sha) [inline packages/polar/src/types.ts:377]: can we remove these types ? or these will be used later ?

File: packages/polar/src/lib/deploy/contract.ts [hunk 6]
```diff
@@ -9,60 +9,173 @@ import {
...
+  }
+
+  async parseSchema (): Promise<void> {
+    await this.queryAbi.parseSchema();
+    await this.executeAbi.parseSchema();
 
-    if (!fs.existsSync(this.contractPath)) {
-      throw new PolarError(ERRORS.ARTIFACTS.NOT_FOUND, {
-        param: this.contractName
-      });
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:40]: where are we checking if `user hasn't compiled contract we will compile it with a warning` ?

File: packages/polar/src/lib/deploy/contract.ts (hunk 4 above)
Comment (udit-gulati) [reply packages/polar/src/lib/deploy/contract.ts:122]: It's only warning cause contract object can be created without schema but query and execute wont be available

**Review by coco-sha — APPROVED** [review]
Looks good.

File: packages/polar/src/lib/deploy/contract.ts (hunk 6 above)
Comment (udit-gulati) [reply packages/polar/src/lib/deploy/contract.ts:40]: In compress function

File: packages/polar/src/lib/deploy/contract.ts (hunk 3 above)
Comment (udit-gulati) [reply packages/polar/src/lib/deploy/contract.ts:108]: TODO later

**Pull request merged**